exporters/trace/zipkin/src/main/java/io/opencensus/exporter/trace/zipkin/ZipkinExporterConfiguration.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright 2019, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.exporter.trace.zipkin;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import io.opencensus.common.Duration;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+import zipkin2.codec.SpanBytesEncoder;
+import zipkin2.reporter.Sender;
+
+/**
+ * Configurations for {@link ZipkinTraceExporter}.
+ *
+ * @since 0.22
+ */
+@AutoValue
+@Immutable
+public abstract class ZipkinExporterConfiguration {
+
+  @VisibleForTesting static final Duration DEFAULT_DEADLINE = Duration.create(10, 0);
+
+  ZipkinExporterConfiguration() {}
+
+  /**
+   * Returns the service name.
+   *
+   * @return the service name.
+   * @since 0.22
+   */
+  public abstract String getServiceName();
+
+  /**
+   * Returns the Zipkin V2 URL.
+   *
+   * @return the Zipkin V2 URL.
+   * @since 0.22
+   */
+  public abstract String getV2Url();
+
+  /**
+   * Returns the Zipkin sender.
+   *
+   * @return the Zipkin sender.
+   * @since 0.22
+   */
+  @Nullable
+  public abstract Sender getSender();
+
+  /**
+   * Returns the {@link SpanBytesEncoder}.
+   *
+   * <p>Default is {@link SpanBytesEncoder#JSON_V2}.
+   *
+   * @return the {@code SpanBytesEncoder}
+   * @since 0.22
+   */
+  public abstract SpanBytesEncoder getEncoder();
```

Comment:
This is a little surprising to me - why is it necessary here?

Reply:
The encoder has been a required field before:
https://github.com/census-instrumentation/opencensus-java/blob/1dc78a87748ed5642ff560ab0ed35c2f1021170d/exporters/trace/zipkin/src/main/java/io/opencensus/exporter/trace/zipkin/ZipkinExporterHandler.java#L68

It's used here:
https://github.com/census-instrumentation/opencensus-java/blob/1dc78a87748ed5642ff560ab0ed35c2f1021170d/exporters/trace/zipkin/src/main/java/io/opencensus/exporter/trace/zipkin/ZipkinExporterHandler.java#L199